src/blocks/test/e2e/performance/performance.spec.js

```diff
@@ -0,0 +1,319 @@
+/**
+ * External dependencies
+ */
+import { basename, join } from 'path';
+import { writeFileSync } from 'fs';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	createNewPost,
+	saveDraft,
+	publishPost,
+	insertBlock,
+	openGlobalBlockInserter,
+	closeGlobalBlockInserter,
+	openListView,
+	closeListView
+} from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	readFile,
+	deleteFile,
+	getTypingEventDurations,
+	getClickEventDurations,
+	getHoverEventDurations,
+	getSelectionEventDurations,
+	getLoadingDurations
+} from './utils';
+
+jest.setTimeout( 1000000 );
+
+describe( 'Post Editor Performance', () => {
+	const results = {
+		serverResponse: [],
+		firstPaint: [],
+		domContentLoaded: [],
+		loaded: [],
+		firstContentfulPaint: [],
+		firstBlock: [],
+		type: [],
+		focus: [],
+		listViewOpen: [],
+		inserterOpen: [],
+		inserterHover: [],
+		inserterSearch: []
+	};
+	const traceFile = __dirname + '/trace.json';
+	let traceResults;
+
+	beforeAll( async() => {
+		const html = readFile(
+			join( __dirname, '/assets/large-otter-post.html' )
+		);
+
+		await createNewPost();
+		await page.evaluate( ( _html ) => {
+			const { parse } = window.wp.blocks;
+			const { dispatch } = window.wp.data;
+			const blocks = parse( _html );
+
+			blocks.forEach( ( block ) => {
+				if ( 'core/image' === block.name ) {
+					delete block.attributes.id;
+					delete block.attributes.url;
+				}
+			});
+
+			dispatch( 'core/block-editor' ).resetBlocks( blocks );
+		}, html );
+
+		await saveDraft();
+	});
+
+	afterAll( async() => {
+		const resultsFilename = basename( __filename, '.js' ) + '.results.json';
+		writeFileSync(
+			join( __dirname, resultsFilename ),
+			JSON.stringify( results, null, 2 )
+		);
+		deleteFile( traceFile );
+	});
+
+	beforeEach( async() => {
+
+		// Disable auto-save to avoid impacting the metrics.
+		await page.evaluate( () => {
+			window.wp.data.dispatch( 'core/editor' ).updateEditorSettings({
+				autosaveInterval: 100000000000,
+				localAutosaveInterval: 100000000000
+			});
+		});
+	});
+
+	// it( 'Loading', async() => {
+
+	// 	// Measuring loading time.
+	// 	let i = 5;
+	// 	while ( i-- ) {
+	// 		if ( await page.$( '.editor-post-save-draft' ) ) {
+	// 			await saveDraft();
+	// 		}
+	// 		await page.reload();
+
+	// 		await page.waitForSelector( '.wp-block' );
+	// 		const {
+	// 			serverResponse,
+	// 			firstPaint,
+	// 			domContentLoaded,
+	// 			loaded,
+	// 			firstContentfulPaint,
+	// 			firstBlock
+	// 		} = await getLoadingDurations();
+
+	// 		results.serverResponse.push( serverResponse );
+	// 		results.firstPaint.push( firstPaint );
+	// 		results.domContentLoaded.push( domContentLoaded );
+	// 		results.loaded.push( loaded );
+	// 		results.firstContentfulPaint.push( firstContentfulPaint );
+	// 		results.firstBlock.push( firstBlock );
+	// 	}
+	// });
+
+	it( 'Typing', async() => {
+
+		// Measuring typing performance.
+		await insertBlock( 'Paragraph' );
+		let i = 100;
```

Comment:
@HardeepAsrani @selul with this values, the run will for aprox 100 * 2s = 200s. Can we go higher without creating problems for Github Actions limits?

Reply:
Yea, it's fine. Github actions run for minutes, that's normal.